src/doc/algorithm/shrink_bounds.cpp

```diff
@@ -262,10 +262,7 @@ bool shrink_bounds_tilemap(const Image* image,
     shrink = true;
     for (v=bounds.y; v<bounds.y+bounds.h; ++v) {
       const tile_t tile = get_pixel_fast<TilemapTraits>(image, u, v);
-      const tile_t tileIndex = tile_geti(tile);
-      const ImageRef tileImg = tileset->get(tileIndex);
-
-      if (tileImg && !is_plain_image(tileImg.get(), refpixel)) {
+      if (tile != refpixel) {
```

Comment:
Here we are comparing a tile with a pixel color (refpixel can be != 0 in indexed mode when the transparent color != 0)

Test this change with an indexed sprite and transparent mode = 1, e.g.:

![image](https://github.com/aseprite/aseprite/assets/39654/c9ddac0b-8de3-4905-8f15-e972d4428f38)

Reply:
I've looking into this and tested the case you mention, and it doesn't seem to be an issue for tilemaps (please correct me if I'm wrong). As far as I have been able to see, this change affects shrink_bounds_tilemap function, which is called only when the cel's image is from a tilemap layer. Then it doesn't matter if the sprite is indexed, because refpixel takes its value from the cel's image mask color, which still is 0 for the tilemap. 

The only odd thing I saw (but seems to be unrelated to this change, because it also happens on the latest rc) is that the tile with index 0 is painted with the idx-0 color in the tiles palette, instead of the  idx-1. And also after drawing something to generate tiles, I saw that it generates an "empty tile" (actually a tile that is painted with the idx-1 color, which is the current transparent color) to use it instead of the 0:
 <img width="300" alt="Screen Shot 2023-07-19 at 10 54 27" src="https://github.com/aseprite/aseprite/assets/5520828/24ec3c1f-b720-4237-9c61-b4f6e5edf461">

In the other hand, when the transparent color is kept as 0, we got the expected result:
<img width="300" alt="Screen Shot 2023-07-19 at 10 56 11" src="https://github.com/aseprite/aseprite/assets/5520828/421faacb-50c6-41be-a72f-8d68899639c3">